libs/features/mas/commerce/src/wcs.js

```diff
@@ -63,29 +43,35 @@ export function Wcs({ settings }) {
      * If WCS does not provide an offer for particular osi,
      * its pending promise will be rejected with "not found".
      * In case of any other Wcs/Network error, promises are rejected with "bad request".
-     * @param {WcsOptions} options
+     * @param options
      * @param {WcsPromises} promises
      * @param reject - used for recursion, prevents rejection of promises with missing offers
      */
     async function resolveWcsOffers(options, promises, reject = true) {
         let message = ERROR_MESSAGE_OFFER_NOT_FOUND;
+        log.debug('Fetching:', options);
         try {
-            log.debug('Fetching:', options);
-            options.offerSelectorIds = options.offerSelectorIds.sort();
-            const { data } = await getWcsOffers(
-                options,
-                {
-                    apiKey,
-                    environment: settings.wcsEnv,
-                    // @ts-ignore
-                    landscape: env === Env.STAGE ? 'ALL' : settings.landscape,
-                },
-                ({ resolvedOffers }) => ({
-                    offers: resolvedOffers.map(applyPlanType),
-                }),
-            );
+          options.offerSelectorIds = options.offerSelectorIds.sort();
+          const url = new URL(settings.wcsURL);
+          url.searchParams.set('offer_selector_ids', options.offerSelectorIds.join(','));
+          url.searchParams.set('country', options.country);
+          url.searchParams.set('language', options.language);
+          url.searchParams.set('locale', options.locale);
+          url.searchParams.set('landscape', env === Env.STAGE ? 'ALL' : settings.landscape);
+          url.searchParams.set('api_key', apiKey);
+          if (options.promotionCode) {
+            url.searchParams.set('promotion_code', options.promotionCode);
+          }
+          if (options.currency) {
+            url.searchParams.set('currency', options.currency);
+          }
+          
+          const response = await fetch(url.toString());
```

Comment:
we need to check this line against network issues.

Reply:
in case of a failed request it falls into a catch clause like before, could you specify what do you mean?

Reply:
When I block the wcs endpoint (screenshot 1), in the before URL, it logs with ERROR_MESSAGE_BAD_REQUEST whereas in the after URL, it doesn't fall in that clause, but is captured by the global catch you added.
So, I wonder what makes it follow a different path in your PR.

Screenshot 1
![image](https://github.com/user-attachments/assets/7859c5e8-9ebc-4d9a-bd34-594d90d9b419)
Screenshot 2
![image](https://github.com/user-attachments/assets/0aec11e1-d162-4cbe-9c98-d802c9f2d9b0)
Screenshot 3
![image](https://github.com/user-attachments/assets/640b414a-809f-410d-a25a-6bf217f133ee)

Reply:
addressed